src/libraries/Microsoft.PowerFx.Interpreter/ReflectionFunction.cs

```diff
@@ -336,19 +337,44 @@ public async Task<FormulaValue> InvokeAsync(IServiceProvider serviceProvider, Fo
                 }
             }
 
-            foreach (var arg in args)
+            List<ErrorValue> errors = null;
+            for (var i = 0; i < args.Length; i++)
```

Comment:
> for (var i = 0; i < args.Length; i++) [](http://example.com/codeflow?start=11&length=38)

nit, 
I see why we switched away from for-each, but consider using locals at the top:

var arg = args[i];
var expectedType = _info.Params[i];